### ISSUE TITLE
Timeout calling the appointments endpoint

### ISSUE THREAD

**Issue by s-finny** [reapit/foundations#6176]
**Describe the bug**
We are calling the appointments endpoint as follows and always get a timeout on the first call.

**To Reproduce**
Endpoint: 
https://platform.reapit.cloud/appointments/?reapit-customer=BEF
 
Payload:
{"officeId":"BIS","marketingMode":"selling","pageSize":100,"modifiedFrom":"2022-02-15T13:25:39.0000000Z","start":"2021-01-01T00:00:00.0000000Z","typeId":["VL","PG","PL","IA"]}
 
Response:
{"message": "Endpoint request timed out"}
 
This is returned after 30 seconds, we retry 60 seconds later and get the results.

**Expected behaviour**
As we are calling this with a modifiedFrom and limiting the typeId we would expect this endpoint to return a small number of results and quite quickly.

Expected result as below, but this still took 18 seconds to respond with nothing.

{
  "_embedded": [],
  "pageNumber": 1,
  "pageSize": 100,
  "pageCount": 0,
  "totalPageCount": 0,
  "totalCount": 0,
  "_links": {
    "self": {
      "href": "/appointments/?PageNumber=1&PageSize=100&marketingMode=selling&modifiedFrom=2022-02-15T13:25:39.0000000Z&officeId=BIS&reapit-customer=BEF&start=2021-01-01T00:00:00.0000000Z&typeId=IA&typeId=VL&typeId=PG&typeId=PL"
    },
    "first": {
      "href": "/appointments/?PageNumber=1&PageSize=100&marketingMode=selling&modifiedFrom=2022-02-15T13:25:39.0000000Z&officeId=BIS&reapit-customer=BEF&start=2021-01-01T00:00:00.0000000Z&typeId=IA&typeId=VL&typeId=PG&typeId=PL"
    }
  }
}

**Comment by plittlewood-rpt** [comment]
Hi @s-finny - thanks for reaching out. Your request looks a little odd as the reapit-customer value should be a header not a query string parameter. In any case, we are already aware of an issue with large databases and specifically using the `modifiedFrom` parameter and it is something we have done some work to resolve already in #5931 however there is some work that needs to be undertaken by our infrastructure team on the database itself before it can be deployed. I will chase our infrastructure team to see where we are it with it and add BEF to the list

**Comment by david-silverink** [comment]
@plittlewood-rpt Receiving a large amount of timeouts today, just to note also.

**Comment by plittlewood-rpt** [comment]
Have chased for update from infrastructure this morning

**Comment by HollyJoyPhillips** [comment]
We've made another request for an update from infrastructure.

**Comment by HollyJoyPhillips** [comment]
FYI, we've requested an update from infrastructure. Will update when possible.

**Comment by HollyJoyPhillips** [comment]
HP to speak with Abdul

**Comment by plittlewood-rpt** [comment]
HP to speak to Abdul

**Comment by plittlewood-rpt** [comment]
This change went into 12.149 which BEF are still not running so we cannot move on this yet. Will speak to internal upgrade team to see where this is at and why it is taking so long

**Comment by plittlewood-rpt** [comment]
@1718066-jpoolton is that any movement on the upgrade for this customer yet?

**Comment by plittlewood-rpt** [comment]
@1718066-jpoolton are we any closer to getting this customer upgraded yet?

**Comment by 1718066-jpoolton** [comment]
Nope, it’s still with Shaz, he was following this up to chase them as had a working UAT for some time. But nothing booked yet.

Sent from Outlook for iOS<https://aka.ms/o0ukef>
________________________________
From: Pete Littlewood ***@***.***>
Sent: Tuesday, June 27, 2023 11:07:20 AM
To: reapit/foundations ***@***.***>
Cc: John Poolton ***@***.***>; Mention ***@***.***>
Subject: [EXTERNAL] Re: [reapit/foundations] Timeout calling the appointments endpoint (Issue #6176)


<https://github.com/1718066-jpoolton>

External Message - Please be cautious when opening links or attachments in email
________________________________
@1718066-jpoolton are we any closer to getting this customer upgraded yet?

**Comment by plittlewood-rpt** [comment]
Customer upgrade is now in the pipeline, after which we can enable the feature flag that will alter the behaviour for this customer

**Comment by plittlewood-rpt** [comment]
Customer has now been upgraded so we can look at altering the Platform configuration. We will need to ensure all rows in the diary table have an amend date. This should be done with webhooks disabled as there are about half a million rows without the info

**Comment by HollyJoyPhillips** [comment]
Waiting for version updates across all DB's from infra. Once confirmed will raise an internal ticket to back date the data.
